test/node-http-test.js

```diff
@@ -0,0 +1,285 @@
+        await mineBlocks(20, cbAddress);
+
+        await wclient.execute('sendopen', [NAME1]);
+        await mineBlocks(1, cbAddress);
+
+        const nameInfo = await nclient.get(`/name/${NAME1}`);
+        assert(nameInfo.start.start < 20);
+        assert.equal(nameInfo.start.reserved, false);
+        assert.equal(nameInfo.info.state, 'OPENING');
+      });
+    });
+  });
+
+  describe('getNameByHash', () => {
+    it('It should return null when an auction has not been initiated', async () => {
+      const nameHash = rules.hashName(NAME0);
+      const name = await nclient.get(`/resource/hash/${nameHash.toString('hex')}`);
+      assert.equal(name, null);
+    });
+
+    describe('When an auction has been initiated', () => {
+      it('It should return the name', async () => {
+        await mineBlocks(10, cbAddress);
+
+        await wclient.execute('sendopen', [NAME0]);
+        await mineBlocks(1, cbAddress);
+
+        const nameHash = rules.hashName(NAME0);
+        const { name } = await nclient.get(`/resource/hash/${nameHash.toString('hex')}`);
+        assert.equal(name, NAME0);
+      });
+    });
+  });
+
+  describe('getNameResource', () => {
+    const records = { compat: false, version: 0, ttl: 172800, ns: ['ns1.example.com.@1.2.3.4'] };
+    it('It should return null when an auction has not been initiated', async () => {
+      const resource = await nclient.get(`/resource/name/${NAME0}`);
+      assert.equal(resource, null);
+    });
+
+    describe('When an auction has been initiated', () => {
+      it('It should return the resource', async () => {
+        await mineBlocks(10, cbAddress);
+
+        await wclient.execute('sendopen', [NAME0]);
+        await mineBlocks(1, cbAddress);
+
+        const { stats: { blocksUntilBidding } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilBidding, cbAddress);
+
+        const sendBid = await wclient.execute('sendbid', [NAME0, 12, 12]);
+        assert(sendBid);
+
+        const { stats: { blocksUntilReveal } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilReveal, cbAddress);
+
+        await wclient.execute('sendreveal', [NAME0]);
+        const { stats: { blocksUntilClose } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilClose, cbAddress);
+
+        await wclient.execute('sendupdate', [NAME0, records]);
+        await mineBlocks(1, cbAddress);
+
+        const resource = await nclient.get(`/resource/name/${NAME0}`);
+        assert.deepEqual(resource, Object.assign(records, { name: NAME0, version: 0 }));
+      });
+    });
+  });
+
+  describe('getNameProof', () => {
+    it('It should return a proof type of TYPE_DEADEND when an auction has not been initiated', async () => {
+      const proof = await nclient.get(`/proof/name/${NAME0}`);
+      assert.equal(proof.type, 'TYPE_DEADEND');
```

Comment:
Nit: assertion against the enums defined here https://github.com/handshake-org/urkel/blob/fbc93f6fbfeabaf15c1bb36b3f5f7d64ee4c240e/lib/radix/proof.js#L664